android/src/BluetoothSensor.java

```diff
@@ -117,6 +117,32 @@ private void enableNotification(BluetoothGattCharacteristic c) {
     }
   }
 
+  /**
+   * Data in the characteristic has little endian byteorder.
+   * Lowest bit of flags indicates valid revs_per_sec reading.
+   * 0 Kelvin indicates invalid temperatures e.g 
+   * no temperature sensor present.
+  */
+  private void engineSensorDataToListeners(BluetoothGattCharacteristic c) {
+    final int flags = c.getIntValue(c.FORMAT_UINT8, 0);
+    final int cht_temp = c.getIntValue(c.FORMAT_UINT16, 1);
+    final int egt_temp = c.getIntValue(c.FORMAT_UINT16, 3);
+    final int outside_air_temperature = c.getIntValue(c.FORMAT_UINT16, 5);
+    if(outside_air_temperature != 0)
```

Comment:
... and when you're fixing coding style: my eyes hurt when I see `if(` without space ... `if` is not a function.